src/main/java/com/google/api/codegen/transformer/ModelTypeNameConverter.java

```diff
@@ -15,52 +15,99 @@
 import com.google.api.tools.framework.model.ProtoElement;
 import com.google.api.tools.framework.model.TypeRef;
 
 /** ModelTypeNameConverter maps TypeRef instances to TypeName instances. */
-public interface ModelTypeNameConverter {
+public abstract class ModelTypeNameConverter implements TypeNameConverter {
   /** Provides a TypeName for the given TypeRef. */
-  TypeName getTypeName(TypeRef type);
+  public abstract TypeName getTypeName(TypeRef type);
 
   /** Provides a TypedValue for the given enum TypeRef. */
-  TypedValue getEnumValue(TypeRef type, EnumValue value);
+  public abstract TypedValue getEnumValue(TypeRef type, EnumValue value);
 
   /** Provides a TypeName for the element type of the given TypeRef. */
-  TypeName getTypeNameForElementType(TypeRef type);
+  public abstract TypeName getTypeNameForElementType(TypeRef type);
 
   /** Provides a TypeName for the given FieldConfig and resource short name. */
-  TypeName getTypeNameForTypedResourceName(FieldConfig fieldConfig, String typedResourceShortName);
+  public abstract TypeName getTypeNameForTypedResourceName(
+      FieldConfig fieldConfig, String typedResourceShortName);
 
   /**
    * Provides a TypeName for the given FieldConfig and resource short name, using the inner type if
    * the underlying field is repeated.
    */
-  TypeName getTypeNameForResourceNameElementType(
+  public abstract TypeName getTypeNameForResourceNameElementType(
       FieldConfig fieldConfig, String typedResourceShortName);
 
   /** Provides a TypeName for the given ProtoElement. */
-  TypeName getTypeName(ProtoElement elem);
+  public abstract TypeName getTypeName(ProtoElement elem);
 
   /** Provides a TypeName for the given short name, using the default package. */
-  TypeName getTypeNameInImplicitPackage(String shortName);
+  public abstract TypeName getTypeNameInImplicitPackage(String shortName);
 
   /**
    * Provides a TypedValue containing the zero value of the given type, plus the TypeName of the
    * type; suitable for use within code snippets.
    */
-  TypedValue getSnippetZeroValue(TypeRef type);
+  public abstract TypedValue getSnippetZeroValue(TypeRef type);
 
   /**
    * Provides a TypedValue containing the zero value of the given type, for use internally within
    * the vkit layer; plus the TypeName of the type. This will often return the same value as {@link
    * #getSnippetZeroValue(TypeRef)}.
    */
-  TypedValue getImplZeroValue(TypeRef type);
+  public abstract TypedValue getImplZeroValue(TypeRef type);
+
+  /** Renders the given value if it is a primitive type. */
+  public abstract String renderPrimitiveValue(TypeRef type, String value);
+
+  // Overriden interface methods call the overloaded method on the underlying Field object.
+
+  /** Provides a TypeName for the given FieldType. */
+  @Override
+  public TypeName getTypeName(FieldType type) {
+    return getTypeName(type.getProtoBasedField().getType());
+  }
```

Comment:
Could you explain that comment a little more?

TypeNameConverter is the root interface for all TypeName util classes, and it only accepts generalized types. 
ModelTypeNameConverter bridges the gap between model types and the generalized types.
SchemaTypeNameConverter bridges the gap between discovery-specific types and the generalized types.